src/test/java/com/exasol/spark/s3/S3DataReadingIT.java

```diff
@@ -1,35 +1,134 @@
+        final MapFunction<Row, Integer> doubleValue = row -> row.getInt(0) * 2;
+        final Dataset<Integer> df = spark.read() //
+                .format(this.format) //
+                .option("table", table.getFullyQualifiedName()) //
+                .options(getSparkOptions()) //
+                .load() //
+                .map(doubleValue, Encoders.INT());
+        assertThat(df.collectAsList(), contains(2, 4, 6, 8, 10, 12));
+    }
+
+    @Test
+    void testMapPartitionsTransformation() {
+        final MapPartitionsFunction<Row, String> convertToString = it -> {
+            final List<String> result = new ArrayList<>();
+            while (it.hasNext()) {
+                result.add(String.valueOf(it.next().getInt(0)));
+            }
+            return result.iterator();
+        };
+        final Dataset<String> df = spark.read() //
+                .format(this.format) //
+                .option("table", table.getFullyQualifiedName()) //
+                .options(getSparkOptions()) //
+                .load() //
+                .mapPartitions(convertToString, Encoders.STRING());
+        assertThat(df.collectAsList(), contains("1", "2", "3", "4", "5", "6"));
+    }
+
+    @Test
+    void testFlatMapTransformation() {
+        final FlatMapFunction<Row, Integer> repeatItem = row -> Arrays.asList(row.getInt(0), row.getInt(0)).iterator();
+        final Dataset<Integer> df = spark.read() //
+                .format(this.format) //
+                .option("table", table.getFullyQualifiedName()) //
+                .options(getSparkOptions()) //
+                .load() //
+                .flatMap(repeatItem, Encoders.INT());
+        assertThat(df.collectAsList(), contains(1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6));
+    }
+
+    @Test
+    void testFilterTransformation() {
+        final FilterFunction<Row> keepEvenNumbers = row -> (row.getInt(0) % 2) == 0 ? true : false;
+        final Dataset<Integer> df = spark.read() //
+                .format(this.format) //
+                .option("table", table.getFullyQualifiedName()) //
+                .options(getSparkOptions()) //
+                .load() //
+                .filter(keepEvenNumbers) //
+                .map((MapFunction<Row, Integer>) row -> row.getInt(0), Encoders.INT());
+        assertThat(df.collectAsList(), contains(2, 4, 6));
+    }
+
+    @Test
+    void testThrowsIfNumberOfPartitionsExceedsMaximumAllowed() {
+        final Dataset<Row> df = spark.read() //
+                .format(this.format) //
+                .option("table", table.getFullyQualifiedName()) //
+                .options(getSparkOptions()) //
+                .option("numPartitions", "1001") //
+                .load();
+        final ExasolValidationException exception = assertThrows(ExasolValidationException.class,
+                () -> df.collectAsList());
+        assertThat(exception.getMessage(), containsString("exceeds the supported maximum of 1000."));
```

Comment:
my guess is, you choose this value as an arbitrary limit.

Reply:
Ohh now I don't remember, but I think there was reason for it. It is either because of CSV loader, or pagination from S3 listing for more than 1000 files. 

I have updated #162 to document the reason.